### PR TITLE
DataPreservers for Starlight runtime

### PR DESCRIPTION
**Pull request by fgamundi**
Adds `DataPreservers` pallet to Starlight runtime.

### CODE REVIEW

**Comment by girazoki** [comment]
Typescript tests please :)

File: solo-chains/runtime/starlight/src/tests/integration_test.rs [hunk 1]
```diff
@@ -276,3 +276,34 @@ fn test_configuration_on_session_change() {
         assert_eq!(CollatorConfiguration::config().collators_per_container, 10);
     });
 }
+
+#[test]
```
Comment (girazoki) [inline solo-chains/runtime/starlight/src/tests/integration_test.rs:280]: Add more tests related to:

- creation of profile
- assignation to a chain
- root can assign to a chain forcedly
- on deregistration, what happens with assignation

File: solo-chains/runtime/starlight/src/lib.rs [hunk 2]
```diff
@@ -1555,7 +1675,7 @@ impl pallet_registrar::RegistrarHooks for StarlightRegistrarHooks {
                 .expect("to fit in BoundedVec"),
             para_ids: ParaIdsFilter::AnyParaId,
             mode: ProfileMode::Bootnode,
-            assignment_request: PreserversAssignementPaymentRequest::Free,
+            assignment_request: PreserversAssignmentPaymentRequest::Free,
```
[the remark this answers is not in the document]
Comment (fgamundi) [reply solo-chains/runtime/starlight/src/lib.rs:1675]: fixed

File: solo-chains/runtime/starlight/src/tests/integration_test.rs (hunk 1 above)
Comment (fgamundi) [reply solo-chains/runtime/starlight/src/tests/integration_test.rs:280]: Those cases are covered by the TS test that I've added. Should I also add equivalents for integration tests?

**Review by nanocryk — APPROVED** [review]

File: solo-chains/runtime/starlight/src/tests/integration_test.rs (hunk 1 above)
Comment (girazoki) [reply solo-chains/runtime/starlight/src/tests/integration_test.rs:280]: I would rather have duplicated versions, specially if those tests are already in dancebox as rust integration tests too

Comment (girazoki) [reply solo-chains/runtime/starlight/src/tests/integration_test.rs:281]: Fixed

**Pull request merged**